src/dialog-controller.js

```diff
@@ -73,7 +75,10 @@ export class DialogController {
           });
       }
 
-      return Promise.resolve();
```

Comment:
I think so. At least I do not see any meaning.
 1.  we are in the success callback
 2. `canDeactivate` is `false`
 3. we return
  - no `return` statement
  - `return Promise.resolve()` 
 4. `_closePromise` will be a promise that is resolved without a result